lib/ansible/modules/cloud/amazon/ec2_tag.py

```diff
@@ -167,8 +167,9 @@ def main():
             for (key, value) in set(tags.items()):
                 if (key, value) not in set(tagdict.items()):
                     dictadd[key] = value
-        tagger = ec2.create_tags(resource, dictadd)
-        gettags = ec2.get_all_tags(filters=filters)
+        if not module.check_mode:
+            tagger = ec2.create_tags(resource, dictadd)
+            gettags = ec2.get_all_tags(filters=filters)
```

Comment:
So...

Line 152 - Get all the remote tags from Ec2
Line 171 - Create the new tags on Ec2
Line 172 - Gets the new, updated tag list from Ec2

Since the tag list is not updated with check_mode I don't need to get all the tags again and I can just skip the step. Line 172 was already in the module so I didn't remove it, but I understand that it is not really useful since the variable `gettags` is not used anywhere after that. I will remove it.